issue_13/script.js

```diff
@@ -0,0 +1,86 @@
+const modalBtnWrapper = document.getElementById('js-modal-btn__wrapper');
+const modalBtn = document.getElementById('js-modal-btn');
+const modalWrapper = document.getElementById('js-modal__wrapper');
+const modalBack = document.getElementById('js-modal-back');
+const closeBtn = document.getElementById('js-close-btn');
+const requestBtn = document.getElementById('js-request-btn');
+const modalButtonElements = [modalBtn, modalBack, closeBtn];
+
+for (let i = 0; i < modalButtonElements.length; i++) {
+    modalButtonElements[i].addEventListener('click', () => {
+        modalBtnWrapper.classList.toggle('hide');
+        modalWrapper.classList.toggle('show');
+    });
+}
+
+requestBtn.addEventListener('click', () => {
+    init();
+    modalWrapper.remove();
+});
+
+function startLoading() {
+    const loadingImg = document.createElement('img');
+    loadingImg.src = 'loading-circle.gif';
+    loadingImg.alt = 'ローディング画像';
+    document.getElementById('js-loading').appendChild(loadingImg);
+}
+
+function removeLoading() {
+    document.getElementById('js-loading').remove();
+}
+
+function renderList(resolvedData) {
+    const ul = document.getElementById('js-ul');
+    const fragment = document.createDocumentFragment();
+    for (const item of resolvedData) {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+
+        a.href = item.a;
+        a.textContent = item.text;
+        img.src = item.img;
+        img.alt = item.alt;
+
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+    ul.appendChild(fragment);
+}
+
+function renderErrorMessage(messageText) {
+    document.getElementById('js-ul').textContent = messageText;
+}
+
+async function fetchData() {
+    try {
+        const url = 'https://api.json-generator.com/templates/NluaaELSLhVe/data?access_token=ambjpfxjl00e50wa639kwndq1ofq3iuykdrv98ge';
+        const response = await fetch(url);
+        if (!response.ok) {
+            console.error(`${response.status}:${response.statusText}`);
+            renderErrorMessage(`${response.status}:${response.statusText}`);
+        }
+        const result = await response.json();
+        return result.data;
+    } catch (error) {
+        console.error(error);
+        renderErrorMessage(`${error.name}:${error.message}`);
+    }
+}
+
+async function init() {
+    startLoading();
+    let listData;
+    try {
+        listData = await fetchData();
+    } catch (error) {
+        console.error(error.message);
+        renderErrorMessage(error.message);
```

Comment:
IMO:
I have added the following code to prevent further processing from working when a "40x or 50x"server error occurs.

  ```
try {
    listData = await fetchData();
    console.log(listData);
    if (!listData) {      //Added code
      return;
    }
  } catch (error) {

```

Reply:
Thank you for your comment.
I had a prior error that was affecting the code behind it, but the code you suggested solved it. :raised_hands:
https://github.com/yoshikawa-akr/js-lessons/pull/13/commits/3e0e51a054db5f1ceac07b6bf455fab3a72fba13